src/ua_timer.c

```diff
@@ -112,10 +114,30 @@ UA_Timer_addRepeatedCallback(UA_Timer *t, UA_ApplicationCallback callback,
     if(interval == 0)
         return UA_STATUSCODE_BADINTERNALERROR;
 
-    UA_DateTime nextTime = UA_DateTime_nowMonotonic() + (UA_DateTime)interval;
+    UA_DateTime currentTime = UA_DateTime_nowMonotonic();
+    UA_DateTime nextTime;
+    if(baseTime <= 0.0) {
```

Comment:
UA_DateTime is a signed integer.
Please don't rely on 0 as a special date value.